crates/starknet/src/starknet/mod.rs

```diff
@@ -159,6 +175,95 @@ impl Starknet {
 
         this.restart_pending_block()?;
 
+        // Re-execute transactions
+        if transactions.is_some() {
```

Comment:
This definitely has to be different function

Reply:
Fixed in 27ec8cc63261fb9430c873a0adda2ceebf38f005 but it will be refactored later as separate module

Reply:
fixed also in 42f064dcf59fabf0427e179c8909cc3d6709b204